### PR TITLE
ONEAPP-2033 Add fingerprint-level ocfDeviceTypes

### PR DESCRIPTION
**Pull request by greens**
ICP-3935
ICP-3010
ICP-4151
ICP-4325
ICP-4579
ICP-4612
ICP-4615
ICP-4685
ICP-4488
ICP-5459
ICP-7011
ICP-7099
ICP-7442
ICP-7507
ICP-8147
ICP-6456
ICP-10197
ICP-8602
ICP-5569
ICP-3821
ICP-3781
ICP-3967
ICP-9498
ICP-9576
ICP-10221

Adds ocfDeviceType to several fingerprints on DTHs that have different icons at a device-level

### CODE REVIEW

**Comment by greens** [comment]
Who's excited?

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 1]
```diff
@@ -27,21 +27,21 @@ metadata {
 
 		fingerprint inClusters: "0x25,0x32"
 		fingerprint mfr: "0086", prod: "0003", model: "0012", deviceJoinName: "Aeotec Micro Smart Switch"
-		fingerprint mfr: "021F", prod: "0003", model: "0087", deviceJoinName: "Dome On/Off Plug-in Switch"
-		fingerprint mfr: "0086", prod: "0103", model: "0060", deviceJoinName: "Aeotec Smart Switch 6"  //US
-		fingerprint mfr: "0086", prod: "0003", model: "0060", deviceJoinName: "Aeotec Smart Switch 6"  //EU
+		fingerprint mfr: "021F", prod: "0003", model: "0087", deviceJoinName: "Dome On/Off Plug-in Switch", , ocfDeviceType: "oic.d.smartplug"
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Unfortunately, this won't compile with the ", ," after the deviceJoinName.

**Review by dkirker — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy (hunk 1 above)
Comment (greens) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: figured I made an oopsie like that. thanks for catching it

Fixed

**Review by dkirker — DISMISSED** [review]

File: devicetypes/smartthings/zigbee-dimmer.src/zigbee-dimmer.groovy [hunk 2]
```diff
@@ -65,7 +65,7 @@ metadata {
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0300, 0702, 0B05", outClusters: "0019", manufacturer: "sengled", model: "E1A-AC2", deviceJoinName: "Sengled DownLight"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-dimmer.src/zigbee-dimmer.groovy:65]: I'm actually commenting on the line above. I think `"Aurora Smart Rotary Dimmer"` should be a switch too

File: devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy [hunk 3]
```diff
@@ -25,17 +25,17 @@ metadata {
...
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006", outClusters: "0003, 0006, 0008, 0019, 0406", manufacturer: "Leviton", model: "DL15A", deviceJoinName: "Leviton Lumina RF Plug-In Appliance Module"
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006", outClusters: "0003, 0006, 0008, 0019, 0406", manufacturer: "Leviton", model: "DL15S", deviceJoinName: "Leviton Lumina RF Switch"
-		fingerprint profileId: "C05E", inClusters: "0000, 0003, 0004, 0005, 0006, 1000, 0B04, FC0F", outClusters: "0019", manufacturer: "OSRAM", model: "Plug 01", deviceJoinName: "OSRAM SMART+ Plug"
+		fingerprint profileId: "C05E", inClusters: "0000, 0003, 0004, 0005, 0006, 1000, 0B04, FC0F", outClusters: "0019", manufacturer: "OSRAM", model: "Plug 01", deviceJoinName: "OSRAM SMART+ Plug", ocfDeviceType: "oic.d.smartplug"
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0B05, FC01, FC08", outClusters: "0003, 0019", manufacturer: "LEDVANCE", model: "PLUG", deviceJoinName: "SYLVANIA SMART+ Smart Plug"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy]: This and a few others in here are also outlets

File: devicetypes/smartthings/zwave-dimmer-switch-generic.src/zwave-dimmer-switch-generic.groovy [hunk 4]
```diff
@@ -24,28 +24,28 @@ metadata {
 
 		fingerprint inClusters: "0x26", deviceJoinName: "Z-Wave Dimmer"
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-dimmer-switch-generic.src/zwave-dimmer-switch-generic.groovy]: Are any of the fingerprints in here actually lights? I wonder if we should just change the `ocfDeviceType` at the DTH level to `"oic.d.switch"` and then override the few smart plugs

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 5]
```diff
@@ -27,21 +27,21 @@ metadata {
...
 		fingerprint mfr: "0086", prod: "0003", model: "0012", deviceJoinName: "Aeotec Micro Smart Switch"
-		fingerprint mfr: "021F", prod: "0003", model: "0087", deviceJoinName: "Dome On/Off Plug-in Switch"
-		fingerprint mfr: "0086", prod: "0103", model: "0060", deviceJoinName: "Aeotec Smart Switch 6"  //US
-		fingerprint mfr: "0086", prod: "0003", model: "0060", deviceJoinName: "Aeotec Smart Switch 6"  //EU
+		fingerprint mfr: "021F", prod: "0003", model: "0087", deviceJoinName: "Dome On/Off Plug-in Switch", ocfDeviceType: "oic.d.smartplug"
+		fingerprint mfr: "0086", prod: "0103", model: "0060", deviceJoinName: "Aeotec Smart Switch 6", ocfDeviceType: "oic.d.smartplug"  //US
+		fingerprint mfr: "0086", prod: "0003", model: "0060", deviceJoinName: "Aeotec Smart Switch 6", ocfDeviceType: "oic.d.smartplug"  //EU
 		fingerprint mfr: "0086", prod: "0103", model: "0074", deviceJoinName: "Aeotec Nano Switch"
 		fingerprint mfr: "0086", prod: "0003", model: "0074", deviceJoinName: "Aeotec Nano Switch"
 		fingerprint mfr: "014F", prod: "574F", model: "3535", deviceJoinName: "GoControl Wall-Mounted Outlet"
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: This looks like it should be a plug

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 6]
```diff
@@ -27,21 +27,21 @@ metadata {
...
 		fingerprint mfr: "014F", prod: "5053", model: "3531", deviceJoinName: "GoControl Plug-in Switch"
 		fingerprint mfr: "0063", prod: "4F44", model: "3031", deviceJoinName: "GE Direct-Wire Outdoor Switch"
-		fingerprint mfr: "0258", prod: "0003", model: "0087", deviceJoinName: "NEO Coolcam Power plug"
+		fingerprint mfr: "0258", prod: "0003", model: "0087", deviceJoinName: "NEO Coolcam Power plug", ocfDeviceType: "oic.d.smartplug"
 		fingerprint mfr: "010F", prod: "0602", model: "1001", deviceJoinName: "Fibaro Wall Plug ZW5" // EU
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Some more plugs

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 7]
```diff
@@ -27,21 +27,21 @@ metadata {
...
 		fingerprint mfr: "010F", prod: "1801", model: "1000", deviceJoinName: "Fibaro Wall Plug ZW5" // UK
 		fingerprint mfr: "0086", prod: "0003", model: "004E", deviceJoinName: "Aeotec Heavy Duty Smart Switch" //EU
 		fingerprint mfr: "0086", prod: "0103", model: "004E", deviceJoinName: "Aeotec Heavy Duty Smart Switch" //US
 		//zw:L type:1001 mfr:0258 prod:0003 model:1087 ver:3.94 zwv:4.05 lib:03 cc:5E,72,86,85,59,5A,73,70,25,27,71,32,20 role:05 ff:8700 ui:8700
-		fingerprint mfr: "0258", prod: "0003", model: "1087", deviceJoinName: "NEO Coolcam Power Plug"  //EU
+		fingerprint mfr: "0258", prod: "0003", model: "1087", deviceJoinName: "NEO Coolcam Power Plug", ocfDeviceType: "oic.d.smartplug"  //EU
 		fingerprint mfr: "027A", prod: "0101", model: "000D", deviceJoinName: "Zooz Power Switch"
 		fingerprint mfr: "0159", prod: "0002", model: "0054", deviceJoinName: "Qubino Smart Plug"
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Another plug

File: devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy [hunk 8]
```diff
@@ -25,17 +25,17 @@ metadata {
...
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0019", manufacturer: "Third Reality, Inc", model: "3RSS008Z", deviceJoinName: "RealitySwitch Plus"
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0019", manufacturer: "Third Reality, Inc", model: "3RSS007Z", deviceJoinName: "RealitySwitch"
-		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0B05", outClusters: "0003, 0006, 0019", manufacturer: "Centralite Systems", model: "4200-C", deviceJoinName: "Centralite Smart Outlet"
+		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0B05", outClusters: "0003, 0006, 0019", manufacturer: "Centralite Systems", model: "4200-C", deviceJoinName: "Centralite Smart Outlet", ocfDeviceType: "oic.d.smartplug"
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, FC7C", outClusters: "0005, 0019, 0020", manufacturer:"IKEA of Sweden", model: "TRADFRI control outlet", deviceJoinName: "IKEA TRÅDFRI control outlet"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy]: outlet

File: devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy [hunk 9]
```diff
@@ -25,17 +25,17 @@ metadata {
...
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0006, 0402, 0B05, FC01, FC02", outClusters: "0003, 0019", manufacturer: "iMagic by GreatStar", model: "1113-S", deviceJoinName: "Iris Smart Plug"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy]: outlet

File: devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy [hunk 10]
```diff
@@ -25,17 +25,17 @@ metadata {
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0006", outClusters: "000A", manufacturer: "HAI", model: "65A21-1", deviceJoinName: "Leviton Wireless Load Control Module-30amp"
 		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006", outClusters: "0003, 0006, 0008, 0019, 0406", manufacturer: "Leviton", model: "DL15A", deviceJoinName: "Leviton Lumina RF Plug-In Appliance Module"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy]: outlet

File: devicetypes/smartthings/zwave-switch-generic.src/zwave-switch-generic.groovy [hunk 11]
```diff
@@ -24,32 +24,32 @@ metadata {
 		fingerprint inClusters: "0x25", deviceJoinName: "Z-Wave Switch"
 		fingerprint mfr: "001D", prod: "1A02", model: "0334", deviceJoinName: "Leviton Appliance Module"
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-switch-generic.src/zwave-switch-generic.groovy]: I think this is an outlet

**Review by tpmanley — APPROVED** [review]

**Pull request merged**